mh/session.py

```diff
@@ -303,6 +329,17 @@ def get_layer_players(self):
         else:
             assert False, "Can't find layer"
 
+    def shutdown(self, shutdown_type):
```

Comment:
This should be named something more appropriate to what the function is doing, maybe ‘full_layers_end’?